src/lua/error.lua

```diff
@@ -178,11 +178,19 @@ local function error_index(err, key)
     if getter ~= nil then
         return getter(err)
     end
-    local f = ffi.C.error_find_field(err, key)
-    if f ~= nil then
-        return mp_decode(f._data)
+    local method = error_methods[key]
```

Comment:
Why did you move this to the top? This doesn’t look like a backwards-compatible change.

Reply:
I don't know… Moved it back.

Reply:
While we are here, could you please add a test to maintain this masking priority?

Reply:
Done.

Reply:
Given we now allow custom payload fields in eb2c6a4a4de8a958f07e5d4efc4453ded1c3026d it makes sense NOT to mask fields like `unpack` etc. @locker is that correct?

The change can done in a different commit I guess.

Reply:
I think that error object methods shouldn't be masked by payload fields (i.e. they should be checked first in `__index`) because the user can still access masked payload fields through `unpack` while accessing masked methods would be problematic.

Reply:
> error object methods shouldn't be masked by payload fields

Done.

Reply:
Can’t we break a client that used the C error API to set payloads and relied on the masking behavior?